ush/templates/coldstart.yaml

```diff
@@ -0,0 +1,156 @@
+default_task: &default_task
+  attrs:
+    cycledefs: forecast
+    maxtries: '1'
+  nodes: 1:ppn=24
+  walltime: 00:30:00
+  nodesize: "&NCORES_PER_NODE;"
+  entities:
+    - "&RSRV_DEFAULT;"
+  envars: &default_vars
+    GLOBAL_VAR_DEFNS_FP: '&GLOBAL_VAR_DEFNS_FP;'
+    PDY: !cycstr {value: '@Y@m@d'}
+    CDATE: !cycstr {value: '@Y@m@d@H'}
+    CYCLE_DIR: !cycstr {value: '&CYCLE_BASEDIR;/@Y@m@d@H'}
+
+
+task_get_extrn_ics:
+  <<: *default_task
+  entities:
+    - '&RSRV_HPSS;'
+  command: '&LOAD_MODULES_RUN_TASK_FP; "&GET_EXTRN_ICS_TN;" "&JOBSDIR;/JREGIONAL_GET_EXTRN_MDL_FILES'
+  nodes: 1:ppn=1
+  jobname: '&GET_EXTRN_ICS_TN;'
+  join: !cycstr {value: '&LOGDIR;/&GET_EXTRN_ICS_TN;_@Y@m@d@H.log'}
+  envars:
+    <<: *default_vars
+    ICS_OR_LBCS: ICS
+
+task_get_extrn_lbcs:
+  <<: *default_task
+  entities:
+    - '&RSRV_HPSS;'
+  command: '&LOAD_MODULES_RUN_TASK_FP; "&GET_EXTRN_LBCS_TN;" "&JOBSDIR;/JREGIONAL_GET_EXTRN_MDL_FILES'
+  nodes: 1:ppn=1
+  jobname: '&GET_EXTRN_LBCS_TN;'
+  join: !cycstr {value: '&LOGDIR;/&GET_EXTRN_LBCS_TN;_@Y@m@d@H.log'}
+  envars:
+    <<: *default_vars
+    ICS_OR_LBCS: LBCS
+
+
+task_make_ics:
+  <<: *default_task
+  command: '&LOAD_MODULES_RUN_TASK_FP; "&MAKE_ICS_TN;" "&JOBSDIR;/JREGIONAL_MAKE_ICS"'
+  nodes: 4:ppn=12
+  jobname: '&MAKE_ICS_TN;'
+  join: !cycstr {value: '&LOGDIR;/&MAKE_ICS_TN;_@Y@m@d@H.log'}
+  envars:
+    <<: *default_vars
+    SLASH_ENSMEM_SUBDIR: '/'
+  dependency:
+    and: &make_ics_deps
+      taskdep_get_extrn:
+        attrs:
+          task: get_extrn_ics
+      or_grid:
+        datadep_complete:
+          attrs:
+            age: 00:00:00:05
+          text: '&LOGDIR;/&MAKE_GRID_TN;_task_complete.txt'
+        streq:
+          left: '&RUN_TASK_MAKE_GRID;'
+          right: 'FALSE'
+      or_orog:
+        datadep_complete:
+          attrs:
+            age: 00:00:00:05
+          text: '&LOGDIR;/&MAKE_OROG_TN;_task_complete.txt'
+        streq:
+          left: '&RUN_TASK_MAKE_OROG;'
+          right: 'FALSE'
+      or_sfc_climo:
+        datadep_complete:
+          attrs:
+            age: 00:00:00:05
+          text: '&LOGDIR;/&MAKE_SFC_CLIMO_TN;_task_complete.txt'
+        streq:
+          left: '&RUN_TASK_MAKE_SFC_CLIMO;'
+          right: 'FALSE'
```

Comment:
I think this is neat! If I understand correctly, every tag in the dependency dictionary is accounted for recursively, and is split with the first hyphen to get the corresponding rocoto tag, then form the xml statement using additional attributes.

Reply:
That's right.